src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -320,21 +324,33 @@ export class RoslynLanguageServer {
 
     public async openSolution(solutionFile: vscode.Uri): Promise<void> {
         this._solutionFile = solutionFile;
-        await this.sendOpenSolutionNotification();
+        this._projectFiles = new Array<vscode.Uri>();
+        await this.sendOpenSolutionAndProjectsNotifications();
     }
 
-    private async sendOpenSolutionNotification(): Promise<void> {
-        if (
-            this._solutionFile !== undefined &&
-            this._languageClient !== undefined &&
-            this._languageClient.isRunning()
-        ) {
-            const protocolUri = this._languageClient.clientOptions.uriConverters!.code2Protocol(this._solutionFile);
-            await this._languageClient.sendNotification('solution/open', new OpenSolutionParams(protocolUri));
+    public async openProjects(projectFiles: vscode.Uri[]): Promise<void> {
+        this._solutionFile = undefined;
+        this._projectFiles = projectFiles;
+        await this.sendOpenSolutionAndProjectsNotifications();
+    }
+
+    private async sendOpenSolutionAndProjectsNotifications(): Promise<void> {
+        if (this._languageClient !== undefined && this._languageClient.isRunning()) {
+            if (this._solutionFile !== undefined) {
+                const protocolUri = this._languageClient.clientOptions.uriConverters!.code2Protocol(this._solutionFile);
+                await this._languageClient.sendNotification('solution/open', new OpenSolutionParams(protocolUri));
```

Comment:
should this explicitly return so we don't attempt to load both a sln and projects?

Reply:
Both shouldn't be set, but if it was there's no harm: we'll say to open the solution and then also give a set of projects (and in the end we will have loaded the union of everything).